01-ruby-fundamentals/classes-overview.md

```diff
@@ -0,0 +1,84 @@
+
+```ruby
+my_animal = String.new("elephant")
+```
+
+The same is true for arrays and hashes. `[]` is short for `Array.new`, and `{}` for `Hash.new`. `""`, `[]`, and `{}` are syntactic sugar; using _literals_ to _initialize_ a new _instance_ of an _object type_. Most classes do **NOT** have this kind of convenience.
+
+In general, using the syntactic sugar is the right way to go. `"elephant"` is much easier to read than `String.new("elephant")`, and there's rarely a reason to use longer version.
+
+## State and Behavior
+
+Objects are used to tie together _state_ and _behavior_.
+- **State:** data or attributes the object will keep track of
+    - _the letters "a", "b" and "c" in a string `"abc"`_
+- **Behavior:** methods that rely upon or modify the state
+    - _the String `upcase` method_
+
+```ruby
+# state
+my_animal = "elephant"
+your_animal = "giraffe"
+
+# behavior - same method but results vary based on state
+my_animal.upcase # "ELEPHANT
+your_animal.upcase # GIRAFFE
+```
+
+In Ruby, state will be represented using _attributes_, also known as _instance variables_, and behavior will be defined using _methods_.
+
+## What Are Objects Used For?
+
+It can be intimidating to look at `String`, `Hash` or `Array` when learning about objects for the first time. These are fundamental programming concepts that get used in almost every program, how can we ever hope to build or reason about something as complex as that?
+
+Although any abstract idea can be represented as an object, it's sometimes more approachable to think of a physical object to get the idea. Let's imagine a program designed to keep track of a library catalogue, which might have a class called `Book`.
+
+**Question:** what _state_ and _behavior_ might a `Book` class need?
+
+Note that all this state and behavior is for a particular instance of a `Book`. Think about the abstract concept of a book, what is its title or author? That doesn't make any sense. Only when we create an instance do these properties come into focus.
+
+When modeling a real-world object like a book or a person, it's important to remember that you don't need to track everything about it, only what's important to your program. You certainly could include information about the color of the book's cover or how tall it is in your catalogue, but you should only do so if that information is useful to you.
+
+Different kinds of objects have different state and behaviors, and the same real-world object in two different programs might have totally different representations. Think about how different a model of a person would be in a medical application versus in a gradebook. What information is relevant is totally different for the two programs.
```

Comment:
Last sentence needs a question mark.

Reply:
I don't think it does. Maybe it could be reworded to not start with a question word.